### PR TITLE
helper/validation: Add StringMatch

### PR DESCRIPTION
**Pull request by vancluever**
`StringMatch` returns a validation function that can be used to match a
string against a regular expression. This can be used for simple
substring validations or more complex validation scenarios. Optionally,
an error message can be returned so that the user is returned a better
error message other than that their field did not match a regular
expression that they might not be able to understand.

### CODE REVIEW

File: helper/validation/validation.go [hunk 1]
```diff
@@ -106,6 +106,28 @@ func StringLenBetween(min, max int) schema.SchemaValidateFunc {
 	}
 }
 
+// StringMatch returns a SchemaValidateFunc which tests if the provided value
+// matches a given regexp, which must compile or else the function will panic.
```
Comment (jbardin) [inline helper/validation/validation.go]: Looks like the implementation has changed, as it takes a compiled Regexp now.

Comment (vancluever) [reply helper/validation/validation.go]: Right my bad! Will update this shortly

**Comment by vancluever** [comment]
Hey @jbardin finally got around to fixing this! Comment should reflect implementation now

**Comment by vancluever** [comment]
Thanks @jbardin!

**Pull request merged**